### PR TITLE
add .editorconfig file

### PR DESCRIPTION
**Pull request by williamboman**
Makes it easier to comply with the code style guide.

More information about editorconfig can be found [here.](http://editorconfig.org/) Please let me know if I got something wrong.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (vitorgalvao) [inline .editorconfig]: Actually, we might as well remove this. [I started some fixes today](https://github.com/caskroom/homebrew-cask/pulls?q=is%3Apr+is%3Aopen+label%3Adocumentation+author%3Avitorgalvao) and should continue by removing the unnecessary line breaks from the documentation.

**Comment by vitorgalvao** [comment]
@jawshooah @alebcay @Ngrd @fanquake Any objections?

File: .editorconfig (hunk 1 above)
Comment (williamboman) [inline .editorconfig]: Afaik, trailing whitespace can have a meaning in markdown files, which is why explicitly telling editors to trim it is not a good idea.

Comment (vitorgalvao) [inline .editorconfig]: > trailing whitespace can have a meaning in markdown files

We don’t write it with that in mind, so it doesn’t matter. Having trailing whitespace have meaning sounds like one of the worse ideas I’ve come across: it essentially means invisible syntax.

Thank you for making me aware of that, though, will make sure to activate trailing whitespace removal in jave and markdown files in my editor.

Comment (williamboman) [inline .editorconfig]: Don't quote me on that though. I'm too lazy to read through the official spec (is there even one?). I keep seeing this configuration in `.editorconfig` files everywhere, so I think it's correct.

**Comment by alebcay** [comment]
LGTM. :+1:

File: .editorconfig (hunk 1 above)
Comment (vitorgalvao) [inline .editorconfig]: > Don't quote me on that though.

You are correct. Markdown uses two or more spaces at the end of a line to signify a line break.

> I'm too lazy to read through the official spec (is there even one?).

Technically yes, [Gruber’s](https://daringfireball.net/projects/markdown/syntax). In practice there are so many variations that we need to take them case by case.

Our markdown files serve only as documentation, though, and only on Github, so [GitHub Flavored Markdown](https://help.github.com/articles/github-flavored-markdown/) is what matters, and that one [does not need the trailing whitespace](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#line-breaks) (see the technical note), which means we don’t need this. Please remove it.

Comment (williamboman) [inline .editorconfig]: Cool, :+1: done.

**Comment by jawshooah** [comment]
I'm on board, but I'd add this:

```ini
[*.md]
trim_trailing_whitespace = false
```

Markdown uses two trailing spaces to force line breaks within paragraphs.

**Comment by williamboman** [comment]
@jawshooah See https://github.com/caskroom/homebrew-cask/pull/13662#discussion_r38877257

**Comment by Ngrd** [comment]
Everything looks good to me also.

**Comment by jawshooah** [comment]
@williamboman Whoops, didn't see that thread. I'm fine with it, then.

**Comment by vitorgalvao** [comment]
Thank you for the contribution. It was merged directly as 18259219d72a991edcde959b10a555cc9e045127 to keep commit history cleaner. Your contribution is still credited to you.

**Pull request closed without merging**